src/Adapters/Auth/instagram.js

```diff
@@ -4,30 +4,56 @@ const httpsRequest = require('./httpsRequest');
 
 // Returns a promise that fulfills iff this user id is valid.
 function validateAuthData(authData) {
-  return request('users/self/?access_token=' + authData.access_token).then(
-    response => {
-      if (response && response.data && response.data.id == authData.id) {
-        return;
-      }
-      throw new Parse.Error(
-        Parse.Error.OBJECT_NOT_FOUND,
-        'Instagram auth is invalid for this user.'
-      );
+
+    if (authData.api_type != null && authData.api_type == "new_api") {
+
+        return requestNew('me?fields=&access_token=' + authData.access_token).then(
+            response => {
+                if (response && response.data && response.data.id == authData.id) {
+                    return;
+                }
+                throw new Parse.Error(
+                    Parse.Error.OBJECT_NOT_FOUND,
+                    'Instagram auth is invalid for this user.'
+                );
+            }
+        );
+
+    } else {
+
+        return requestOld('users/self/?access_token=' + authData.access_token).then(
+            response => {
+                if (response && response.data && response.data.id == authData.id) {
+                    return;
+                }
+                throw new Parse.Error(
+                    Parse.Error.OBJECT_NOT_FOUND,
+                    'Instagram auth is invalid for this user.'
+                );
+            }
+        );
+
     }
-  );
+
 }
 
 // Returns a promise that fulfills iff this app id is valid.
 function validateAppId() {
-  return Promise.resolve();
+    return Promise.resolve();
+}
+
+// A promisey wrapper for old api requests
+function requestOld(path) {
+    return httpsRequest.get('https://api.instagram.com/v1/' + path);
```

Comment:
We were referring to passing in the api endpoint. This way if a new endpoint is release we won't have to update the server again.

Reply:
The problem is that not everyone is able to know witch endpoint to pass, that solution will be complicated most of devs or just Parse Server lovers.

for example i sell sources code on codecanyon.net and almost 80% of buyers, are not developers, they use to follow documentations to setup their apps.

Reply:
The old api will be defaulted (if no endpoint is passed in). If you want to use a newer endpoint pass it in. This will also enable backwards compatibility.

Reply:
ok